src/runtime/crt/host/microtvm_api_server.py

```diff
@@ -67,6 +88,29 @@ def server_info_query(self, tvm_version):
     # The build target given to make
     BUILD_TARGET = "build/main"
 
+    def _populate_makefile(
+        self,
+        makefile_template_path: pathlib.Path,
+        makefile_path: pathlib.Path,
+        memory_size: int,
```

Comment:
I'd rather this way because it makes it clear at high level APIs, like generate_project, what are the requirement for each option. If we pass `options` to sub-functions, we need to check multiple levels to determine if each option is required/option for each API call.

Reply:
ok